### ISSUE TITLE
Add-on/Scan rule review request

### ISSUE THREAD

**Issue by zapbot** [zaproxy/zaproxy#1558]
```
Summary of changes:
Many of the back end interfaces use a secondary programming language, often compiled,
to interface to a variety of system functions.  C and C++ are two of the more popular
"traditional" languages that are used.  The Format String exploit occurs when the submitted
data of an input string is evaluated as a command by the application. In this way,
the attacker could execute code, read the stack, or cause a segmentation fault in the
running application, causing new behaviors that could compromise the security or the
stability of the system.

The active scanner I have attached injects multiple %s's into a text field.  If it
crashes the back end code (Generally I find 2 to be sufficient) it then reports a format
string vulnerability and cross checks using a %x attack.  The %x attack is interesting
in that it greatly increases the length of the return well out of proportion to the
number of %x's sent.


Build instructions (if non standard):None


Current status (Alpha/Beta/Release):  None


Requested Status (Alpha/Beta/Release):  Alpha



Publish to Marketplace (Y/N):Y


Anything else we should know:


```

Original issue reported on code.google.com by `msrader` on 2015-03-10 14:27:04
<hr>
 * *Attachment: [FormatString.java](https://storage.googleapis.com/google-code-attachments/zaproxy/issue-1558/comment-0/FormatString.java)*

**Comment by msrader** [comment]
Humm, I am trying to add the latest version of the code to this request, and it does not seem to let me.  It will allow me to add images but not .java files.  Any idea what is causing this.

**Comment by thc202** [comment]
Just do a pull request! ;)
It's easier to do the reviews and, finally, add the scanner.

**Comment by msrader** [comment]
Thanks, I will give it a shot later.

**Comment by msrader** [comment]
Well, I did the pull to zap-extensions/Alpha

**Comment by thc202** [comment]
Resolved in zaproxy/zap-extensions@c2555b2400987e84313563381134243f03d080a7
